wp-includes/sqlite-ast/class-wp-sqlite-information-schema-reconstructor.php

```diff
@@ -0,0 +1,427 @@
+		try {
+			$mysql_type = $this->driver->execute_sqlite_query(
+				'SELECT mysql_type FROM _mysql_data_types_cache WHERE `table` = ? AND column_or_index = ?',
+				array( $table_name, $column_or_index_name )
+			)->fetchColumn();
+		} catch ( PDOException $e ) {
+			if ( str_contains( $e->getMessage(), 'no such table' ) ) {
+				return null;
+			}
+			throw $e;
+		}
+		if ( str_ends_with( $mysql_type, ' KEY' ) ) {
+			$mysql_type = substr( $mysql_type, 0, strlen( $mysql_type ) - strlen( ' KEY' ) );
+		}
+		return $mysql_type;
+	}
+
+	/**
+	 * Get a MySQL column type from an SQLite column type.
+	 *
+	 * This method converts an SQLite column type to a MySQL column type as per
+	 * the SQLite column type affinity rules:
+	 *   https://sqlite.org/datatype3.html#determination_of_column_affinity
+	 *
+	 * @param  string $column_type The SQLite column type.
+	 * @return string              The MySQL column type.
+	 */
+	private function get_mysql_data_type( string $column_type ): string {
+		$type = strtoupper( $column_type );
+		if ( str_contains( $type, 'INT' ) ) {
+			return 'int';
+		}
+		if ( str_contains( $type, 'TEXT' ) || str_contains( $type, 'CHAR' ) || str_contains( $type, 'CLOB' ) ) {
+			return 'text';
+		}
+		if ( str_contains( $type, 'BLOB' ) || '' === $type ) {
+			return 'blob';
+		}
+		if ( str_contains( $type, 'REAL' ) || str_contains( $type, 'FLOA' ) ) {
+			return 'float';
+		}
+		if ( str_contains( $type, 'DOUB' ) ) {
+			return 'double';
+		}
+
+		/**
+		 * While SQLite defaults to a NUMERIC column affinity, it's better to use
+		 * TEXT in this case, because numeric SQLite columns in non-strict tables
+		 * can contain any text data as well, when it is not a well-formed number.
+		 *
+		 * See: https://sqlite.org/datatype3.html#type_affinity
+		 */
+		return 'text';
+	}
+
+	/**
+	 * Quote an SQLite identifier.
+	 *
+	 * Wrap the identifier in backticks and escape backtick values within.
+	 *
+	 * @param  string $unquoted_identifier The unquoted identifier value.
+	 * @return string                      The quoted identifier value.
+	 */
+	private function quote_sqlite_identifier( string $unquoted_identifier ): string {
+		return '`' . str_replace( '`', '``', $unquoted_identifier ) . '`';
+	}
+
+	/**
+	 * Unquote a quoted MySQL identifier.
+	 *
+	 * Remove bounding quotes and replace escaped quotes with their values.
+	 *
+	 * @param  string $quoted_identifier The quoted identifier value.
+	 * @return string                    The unquoted identifier value.
+	 */
+	private function unquote_mysql_identifier( string $quoted_identifier ): string {
+		$first_byte = $quoted_identifier[0] ?? null;
+		if ( '"' === $first_byte || '`' === $first_byte ) {
+			$unquoted = substr( $quoted_identifier, 1, -1 );
+		} else {
+			$unquoted = $quoted_identifier;
+		}
+		return str_replace( $first_byte . $first_byte, $first_byte, $unquoted );
```

Comment:
Let's only run that replace in the if branch where that first byte is a quote, otherwise we risk corrupting byte sequences such as `strassberg`

Reply:
Ahh, great catch! The same issue was in the driver for unquoting SQLite (this was copied).